### PR TITLE
upgrade to Pants v2.13.0

### PR DESCRIPTION
**Pull request by tdyas**
Upgrade to Pants v2.13.0 including config changes to opt in to future Pants v2.14.0 behavior.

### CODE REVIEW

File: pants.toml [hunk 1]
```diff
@@ -16,6 +16,11 @@ backend_packages = [
   "pants.backend.experimental.scala.lint.scalafmt",
 ]
 
+# Opt in to future behaviors that will become the default in Pants v2.14.
+# These directives should be removed in Pants v2.14.x and higher.
+use_deprecated_directory_cli_args_semantics = false
```
Comment (stuhood) [inline pants.toml:21]: Ditto readme audit.

File: pants.toml [hunk 2]
```diff
@@ -16,6 +16,11 @@ backend_packages = [
...
+use_deprecated_pex_binary_run_semantics = false
```
Comment (stuhood) [inline pants.toml:22]: This shouldn't be necessary since this repo doesn't use the Python backend... are you seeing otherwise?

**Review by stuhood — APPROVED** [review]

File: pants.toml (hunk 2 above)
Comment (tdyas) [reply pants.toml:22]: Yes the message appeared when I ran `./pants package ::`.

File: pants.toml [hunk 3]
```diff
@@ -50,6 +55,15 @@ args = [
 # Resolve-global `scalac` plugins, which are defined as `scalac_plugin` targets.
 jvm-default = "acyclic"
 
+[scala-infer]
```
Comment (tdyas) [inline pants.toml:58]: Note: The deprecation warning states this section is `scala_infer` (underscore) when in fact it is a `-` dash.

**Pull request merged**